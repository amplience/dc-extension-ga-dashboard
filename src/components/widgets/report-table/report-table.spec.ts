import { tick } from 'svelte';
import { render } from '@testing-library/svelte';
import ReportTable from './report-table.svelte';
import CustomCellMock from './__mocks__/custom-cell-contents-mock.svelte';

let tableConfig;
describe('ReportTable', () => {
  beforeEach(() => {
    tableConfig = {
      columns: [
        {
          title: 'column-title',
          width: '10%',
          align: 'flex-end',
        },
      ],
    };
  });
  it('should render the ReportTable component in a loading state', async () => {
    const { container } = render(ReportTable, {
      data: null,
      config: tableConfig,
    });

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the ReportTable component in a no data state', async () => {
    const { container } = render(ReportTable, {
      data: [],
      config: tableConfig,
    });

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the ReportTable component with data', async () => {
    const { container } = render(ReportTable, {
      data: [['cell-value']],
      config: tableConfig,
    });

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the ReportTable component with data and a custom component', async () => {
    tableConfig.columns[0].component = CustomCellMock;
    const { container } = render(ReportTable, {
      data: [['cell-value']],
      config: tableConfig,
    });

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });
});
