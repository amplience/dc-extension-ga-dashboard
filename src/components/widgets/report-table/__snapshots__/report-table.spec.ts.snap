// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTable should render the ReportTable component in a loading state 1`] = `
<div>
  <div
    class="spinner-container svelte-wkqaim"
    style="--z-index: 10;"
  >
    <div
      class="spinner dot-0 svelte-wkqaim"
    />
    <div
      class="spinner dot-1 svelte-wkqaim"
    />
    <div
      class="spinner dot-2 svelte-wkqaim"
    />
  </div>
  
</div>
`;

exports[`ReportTable should render the ReportTable component in a no data state 1`] = `
<div>
  <div
    class="no-data-container svelte-1kq81j9"
  >
    <p>
      No data for the selected date range
    </p>
  </div>
  
</div>
`;

exports[`ReportTable should render the ReportTable component with data 1`] = `
<div>
  <div
    class="data-table svelte-2e8um5"
  >
    <div
      class="data-table-slot 
    scrollable svelte-2e8um5"
    >
      <div
        class="data-table-head svelte-1nivie3"
      >
        <div
          class="data-table-row svelte-hige1l"
        >
          <div
            class="data-table-cell svelte-1x8v3rk"
            style="
  max-width:10%;
  justify-content:flex-end;
  color:#999;"
          >
            column-title
             
          </div>
          
        </div>
      </div>
       
      <div
        class="data-table-body"
      >
        <div
          class="data-table-row svelte-hige1l"
        >
          <div
            class="data-table-cell svelte-1x8v3rk"
            style="
  max-width:10%;
  justify-content:flex-end;
  color:#333;"
          >
            cell-value
            
          </div>
           
        </div>
        
      </div>
    </div>
  </div>
  
</div>
`;

exports[`ReportTable should render the ReportTable component with data and a custom component 1`] = `
<div>
  <div
    class="data-table svelte-2e8um5"
  >
    <div
      class="data-table-slot 
    scrollable svelte-2e8um5"
    >
      <div
        class="data-table-head svelte-1nivie3"
      >
        <div
          class="data-table-row svelte-hige1l"
        >
          <div
            class="data-table-cell svelte-1x8v3rk"
            style="
  max-width:10%;
  justify-content:flex-end;
  color:#999;"
          >
            column-title
             
          </div>
          
        </div>
      </div>
       
      <div
        class="data-table-body"
      >
        <div
          class="data-table-row svelte-hige1l"
        >
          <div
            class="data-table-cell svelte-1x8v3rk"
            style="
  max-width:10%;
  justify-content:flex-end;
  color:#333;"
          >
            <div>
              0
               
              cell-value
            </div>
            
            
          </div>
           
        </div>
        
      </div>
    </div>
  </div>
  
</div>
`;
